parser.y

```diff
@@ -1268,6 +1271,17 @@ AlterTableSpec:
 			PartitionNames: $4.([]model.CIStr),
 		}
 	}
+|	"EXCHANGE" "PARTITION" Identifier "WITH" "TABLE" TableName WithValidationOpt
```

Comment:
i think it's better that putting `PARTITON` rules into `PartitionOpt`

Reply:
i think it's no necessary to do that. `EXCHANGE PARTITION` is a unique rule so that you cannot reuse it, and the `PARTITION` is not optional.

Reply:
@tangenta PTAL what do you think about that?

Reply:
@kennytm PTAL